src/backend/distributed/commands/dependencies.c

```diff
@@ -31,20 +31,65 @@
 #include "distributed/worker_manager.h"
 #include "distributed/worker_transaction.h"
 
+typedef enum RequiredObjectSet
+{
+	REQUIRE_ONLY_DEPENDENCIES = 1,
+	REQUIRE_OBJECT_AND_DEPENDENCIES = 2,
+} RequiredObjectSet;
+
 
 static void EnsureDependenciesCanBeDistributed(const ObjectAddress *relationAddress);
 static void ErrorIfCircularDependencyExists(const ObjectAddress *objectAddress);
 static int ObjectAddressComparator(const void *a, const void *b);
 static void EnsureDependenciesExistOnAllNodes(const ObjectAddress *target);
+static void EnsureRequiredObjectExistOnAllNodes(const ObjectAddress *target,
```

Comment:
Naming suggestion: Let's use the name of the struct RequiredObjectSet inside the name of this function
```suggestion
static void EnsureRequiredObjectSetExistsOnAllNodes(const ObjectAddress *target,
```

Reply:
the suggestion makes sense to me